src/Sylius/Bundle/ApiBundle/CommandHandler/RegisterShopUserHandler.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\CommandHandler;
+
+use Doctrine\Persistence\ObjectManager;
+use Sylius\Bundle\ApiBundle\Command\RegisterShopUser;
+use Sylius\Component\Core\Model\CustomerInterface;
+use Sylius\Component\Core\Model\ShopUserInterface;
+use Sylius\Component\Core\Repository\CustomerRepositoryInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\Component\User\Canonicalizer\CanonicalizerInterface;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+
+class RegisterShopUserHandler implements MessageHandlerInterface
```

Comment:
I thought about opening  `RegisterShopUser` only. Imho, this class could be finalized